### DISCUSSION TITLE
模板是否应当提供图、表和公式的按章节编号？

### DISCUSSION THREAD

**Discussion by aqni** [Tr0py/NKU-thesis-template-2020#26] [Ideas]
**以下说明针对dev分支中重构后的实现代码**

实现上可能遇到的问题

1. `\thesection`已经被改为中文，`chngcntr`宏包提供的`\counterwithin`命令直接使用会使用中文编号，可以重新实现一个类似`\counterwithin`的命令，不是大问题。
2. 如果提供了图、表和公式的按章编号，那么使用者又使用了算法、伪代码、定理等其他环境的话，其他环境的编号一般默认是全局统一编号，使用者有能力将其他环境的编号改为按章编号吗？注意，教务处格式要求所有编号方式要么全部采用统一编号，要么采用按章编号，不能混用。
3. 在单标题附录中（就是只有一个`附__录`标题），如果附录中使用了相关编号，应该如何编号？

**Comment by Tr0py** [comment]
让我理解下
1. 编号是指`\label{xxx}` 和 `\ref{xxx}` 这样产生的编号，对吧？
2. 你说的教务处格式编号要求是指的[南开大学本科毕业论文（设计）指导手册.doc](http://jwc.nankai.edu.cn/_upload/article/files/6d/c0/d3c02ba64061932efa744dc2ca6d/f891176b-5b72-46de-91f6-bd669276e484.doc) 中的 
> "公式可按全文统编序号，也可按章独立序号，如（49）或（4.11），采用哪一种序号应和图序、表序编法一致。"

**Comment by Tr0py** [comment]
如果是的话，且全局编号是latex默认行为的话，那就保留全局编号最简单。

如果想提供分章节编号，可能需要看下能否实现对于`\label`和`\ref`的重定义？这样可以拦截所有的`\ref`.

**Comment by Tr0py** [comment]
哦我明白了，你说编号的时候除了`\ref` 还有图表公式本身的标题的编号问题。

**Comment by aqni** [comment]
chngcntr 红包提供的`\counterwithin`宏，由于章编号改为中文编号，已经不可用。模板开发者只提供一个可用的`\counterwithin`宏，让有能力的使用者自信修改章节编号吧。

想要一次性将所有编号改为按章节编号似乎不可能。

**Comment by aqni** [comment]
其实最大问题是，如果采取按章节编号，附录中该如何编号，按第几章算？

如果实在没有好的办法，可以在使用者在附录中使用按章节编号时，给出警告或错误。

也可以在单附录标题的情况下，默认使用A作为章编号。

如何？

**Reply by Tr0py** [reply]
我觉得可以。如果格式要求没有明确进行要求，那我们提供任何实现其实都不算错。

不如就用A作为附录的章节编号，并且在我们的文档中对如何修改这一点进行说明。这样可以让我们的模板有正常的分章节编号功能，也可以让有需求的人自己去改。
